vayesta/edmet/edmet.py

```diff
@@ -54,6 +54,17 @@ def eps(self):
         eps = eps.reshape(-1)
         return eps, eps
 
+    @property
+    def e_nonlocal(self):
+        try:
+            return self._e_nonlocal
+        except AttributeError:
+            return 0.0
```

Comment:
as with the `e_rpa`, I think this pattern is bad practice - I think `self._e_nonlocal` should already be initialised as `None`

Reply:
I've added the value into the `_reset` function for EDMET, following the pattern already used for `e_rpa` to initialise it there.

Reply:
yeah I don't like that at all either :smile:

Reply:
oops that comment was meant for the above review - I think that all the variables should be initialised in `__init__` rather than `_reset` even if it duplicates that code, but I guess if the rest of vayesta follows the standard of not doing that then all good

Reply:
Tbf it fitted pretty well with this one as well! Duplication could be justified here- but maybe having a `_initialise_variables` function just under `__init__` might be a reasonable compromise? In any case, I'd suggest this is for a potential future refactor rather than this PR.